### PR TITLE
Updated ant to 1.10.12 for Docker images

### PR DESCRIPTION
**Pull request by azotcsit**
It is required for CASSANDRA-16630. I've tested locally only Ubuntu testing images. If you see no issues with the current approach, I'll get tested the rest.

### CODE REVIEW

File: docker/almalinux-image.docker [hunk 1]
```diff
@@ -20,17 +19,21 @@ RUN yum -y install \
...
-
 RUN pip3 install --upgrade pip
 
 # install Sphinx to generate docs
 RUN pip3 install \
    Sphinx \
    sphinx_rtd_theme
 
+# install ant
+RUN mkdir /usr/share/ant && \
```
Comment (smiklosovic) [inline docker/almalinux-image.docker:30]: I would

1) move ENVs to the top of this docker file, I do not like when ENV's are all over the place
2) I would make environment property out of Ant version too (ANT_VERSION=1.10.12) and I would use it in wget and tar commands. This way, you may test your image with different Ant replacing it at one place or injecting it from outside

https://www.baeldung.com/ops/docker-container-environment-variables

File: docker/almalinux-image.docker [hunk 2]
```diff
@@ -40,7 +43,7 @@ USER build
 
 RUN mkdir -p $RPM_BUILD_DIR/{BUILD,BUILDROOT,RPMS,SOURCES,SPECS,SRPMS}
 
-# Clone Cassandra and cache maven artifacts
+# clone Cassandra and cache maven artifacts
```
Comment (smiklosovic) [inline docker/almalinux-image.docker:46]: maven -> Maven if we want to play this letter capitalisation game.

File: docker/almalinux-image.docker [hunk 3]
```diff
@@ -20,17 +19,21 @@ RUN yum -y install \
...
 RUN pip3 install --upgrade pip
 
 # install Sphinx to generate docs
 RUN pip3 install \
    Sphinx \
    sphinx_rtd_theme
 
+# install ant
+RUN mkdir /usr/share/ant && \
+    wget --tries 9 https://dlcdn.apache.org/ant/binaries/apache-ant-1.10.12-bin.tar.gz && \
```
Comment (smiklosovic) [inline docker/almalinux-image.docker:31]: is this "tries" really needed? The default, according to man, is 20. Do you lower this to 9 on purpose?

You can also download sha512 file (1), do hash of what you downloaded and compare it 

(1) https://dlcdn.apache.org/ant/binaries/apache-ant-1.10.12-bin.tar.gz.sha512

Comment (michaelsembwever) [reply docker/almalinux-image.docker:31]: instead, can we use the `until curl -S -s --retry 9 --retry-connrefused --retry-delay 1 …` approach that already existed (which had evolved from experience)

**Comment by michaelsembwever** [comment]
i've moved the pr branch out of the origin repo
 https://github.com/apache/cassandra-builds/tree/cassandra-16630_ant_version_dockers

**Pull request closed without merging**